src/essreflectometry/types.py

```diff
@@ -0,0 +1,103 @@
+from typing import NewType, TypeVar
+
+import sciline
+import scipp as sc
+
+Reference = NewType('Reference', str)
+Sample = NewType('Sample', str)
+Run = TypeVar('Run', Reference, Sample)
+
+
+class RawData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event time data from nexus file,
+    binned by `detector_number` (pixel of the detector frame)."""
+
+
+class WavelengthData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event data with wavelengths computed for every event,
+    binned by `detector_number` (pixel of the detector frame)"""
+
+
+class ThetaData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event data with wavelengths and scattering angle computed for each event,
+    binned by `detector_number` (pixel of the detector frame)"""
+
+
+class QData(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event data with wavelength, incidence angle and momentum transfer computed per event.
+    Binned by momentum transfer according to the QBins provider,
+    and by `detector_number` (pixel of the detector frame).
+    """
+
+
+class FootprintCorrected(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Event data with weight corrected for the footprint of the beam on the sample for the
+    incidence angle of the event."""
+
+
+SpecularReflectionCoordTransformGraph = NewType(
+    'SpecularReflectionCoordTransformGraph', dict
+)
+
+
+class Histogrammed(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Histogram of event weights by momentum transfer and detector_number."""
+
+
+class Normalized(sciline.Scope[Run, sc.DataArray], sc.DataArray):
+    """Normalization of the histogram.
+    The normalization for the sample consists of scaling
+    with the inverse of the total weight.
+    The normalization for the reference consists of
+        1. multiplication with the calibration factor
+           from the supermirror calibration and
+        2. scaling with the inverse of the total weight.
+    """
+
+
+NormalizedIofQ = NewType('NormalizedIofQ', sc.DataArray)
+'''Normalized histogram over momentrum transfer and detector number,
+normalized by the calibrated reference measurement.'''
+QResolution = NewType('QResolution', sc.Variable)
+'''Resolution term for the momentum transfer for each bin of QBins.'''
+
+
+''' Parameters for the workflow '''
+
+QBins = NewType('QBins', sc.Variable)
+'''Bins for the momentum transfer histogram.'''
+
+
+class Filename(sciline.Scope[Run, str], str):
+    """Filename of the event data nexus file."""
+
+
+# TODO What do they mean?
+# Supermirror parameters
+MValue = NewType('MValue', sc.Variable)
+CriticalEdge = NewType('CriticalEdge', sc.Variable)
+Alpha = NewType('Alpha', sc.Variable)
+SupermirrorCalibrationFactor = NewType('SupermirrorCalibrationFactor', sc.Variable)
```

Comment:
These names make no sense on their own. I think the types should be moved into a supermirror submodule to provide a "namespace".